### ISSUE TITLE
训练时好好的，预测无结果

### ISSUE THREAD

**Issue by Holmeyoung** [MaybeShewill-CV/CRNN_Tensorflow#183]
### 环境
- Python3.6
- CPU
- CentOS7
- tensorflow1.12.0（CPU版）

我是采用 `pip` 方式逐个安装的 `requirements.txt` 里的包

### 训练的结果是收敛的
```sh
python train_shadownet.py --dataset_dir train_space/tfrecords_dir

2018-12-22 16:24:59.372605: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
INFO: 12-22 16:24:59: train_shadownet.py:137 * 139634436175680 Training from scratch
INFO: 12-22 16:25:01: train_shadownet.py:172 * 139634436175680 Epoch: 1 cost= 202.948822
INFO: 12-22 16:25:02: train_shadownet.py:172 * 139634436175680 Epoch: 2 cost= 179.400742
INFO: 12-22 16:25:03: train_shadownet.py:172 * 139634436175680 Epoch: 3 cost= 164.168243
INFO: 12-22 16:25:04: train_shadownet.py:172 * 139634436175680 Epoch: 4 cost= 148.397339
.
.
.
INFO: 12-22 16:26:16: train_shadownet.py:172 * 139634436175680 Epoch: 76 cost= 20.062393
INFO: 12-22 16:26:17: train_shadownet.py:172 * 139634436175680 Epoch: 77 cost= 20.208023
INFO: 12-22 16:26:18: train_shadownet.py:172 * 139634436175680 Epoch: 78 cost= 20.390043
INFO: 12-22 16:26:19: train_shadownet.py:172 * 139634436175680 Epoch: 79 cost= 20.278833
INFO: 12-22 16:26:19: train_shadownet.py:155 * 139634436175680 Cost didn't improve beyond 0.001000 for 7 epochs, stopping early.
Done
```

### 但是执行 `demo_shadownet.py` 预测的时候没有结果
```sh
python demo_shadownet.py --num_classes 0 --image_path train_space/dataset/Train/兰焓.jpg --weights_path model/shadownet/shadownet_2018-12-22-16-27-11.ckpt-399       

/root/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Importing configuration global_configuration.config from .
WARNING:tensorflow:From /root/identity_recog_server/crnn/crnn_model/crnn_model.py:113: BasicLSTMCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is deprecated, please use tf.nn.rnn_cell.LSTMCell, which supports all the feature this cell currently has. Please replace the existing code with tf.nn.rnn_cell.LSTMCell(name='basic_lstm_cell').
WARNING: 12-22 16:35:42: tf_logging.py:125 * 140367175587648 From /root/identity_recog_server/crnn/crnn_model/crnn_model.py:113: BasicLSTMCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is deprecated, please use tf.nn.rnn_cell.LSTMCell, which supports all the feature this cell currently has. Please replace the existing code with tf.nn.rnn_cell.LSTMCell(name='basic_lstm_cell').
2018-12-22 16:35:42.477842: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
INFO:tensorflow:Restoring parameters from model/shadownet/shadownet_2018-12-22-16-27-11.ckpt-399
INFO: 12-22 16:35:42: tf_logging.py:115 * 140367175587648 Restoring parameters from model/shadownet/shadownet_2018-12-22-16-27-11.ckpt-399
INFO: 12-22 16:35:49: demo_shadownet.py:111 * 140367175587648 Predict image 兰焓.jpg label 
```
`lable` 什么也没有

### 我新建的 `train_space` 目录
```sh
.
├── charset_dir
│   ├── char_dict.json
│   └── ord_map.json
├── dataset
│   ├── Test
│   │   ├── sample.txt
│   │   ├── 腌馇.jpg
│   │   ├── 腌筛.jpg
│   │   ├── 锕窘.jpg
│   │   ├── 锕洽.jpg
│   │   ├── 锕偎.jpg
│   │   ├── 阿粝.jpg
│   │   ├── 怍漫.jpg
│   │   ├── 胙党.jpg
│   │   └── 胙狰.jpg
│   └── Train
│       ├── sample.txt
│       ├── 腌谤.jpg
│       ├── 腌孢.jpg
│       ├── 腌埤.jpg
│       ├── 腌奔.jpg
│       ├── 腌铋.jpg
│       ├── 胙斟.jpg
│       ├── 胙涿.jpg
│       ├── 胙浊.jpg
│       └── 胙坐.jpg
├── tfrecords_dir
│   ├── test_feature.tfrecords
│   ├── train_feature.tfrecords
│   └── val_feature.tfrecords
└── tree

5 directories, 998221 files
```
我新建了 `train_space` 文件夹，之后的所有操作都是在这个文件夹完成的
最下面那个 `tree` 文件是我为了生产目录结构生成的文件

### 这是我的 `sample.txt`
```sh
追宽.jpg 追宽
暧邕.jpg 暧邕
寮撬.jpg 寮撬
画珞.jpg 画珞
馓炮.jpg 馓炮
痘忠.jpg 痘忠
袤鹩.jpg 袤鹩
仡阙.jpg 仡阙
畅嘬.jpg 畅嘬
荇鳀.jpg 荇鳀
藠洲.jpg 藠洲
肆鳌.jpg 肆鳌
寤鹌.jpg 寤鹌
洱呃.jpg 洱呃
```
请问是哪里的问题呢
谢谢～

**Comment by Sierkinhane** [comment]
https://github.com/Sierkinhane/crnn_chinese_characters_rec 可以参考一下我的仓库

**Comment by Holmeyoung** [comment]
@Sierkinhane 谢谢啦

**Comment by xiaoaoran** [comment]
I have the same problem.

Importing configuration global_configuration.config from .
2019-01-03 18:51:38.494647: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
INFO:tensorflow:Restoring parameters from model/shadownet/shadownet_2019-01-03-18-33-13.ckpt-399
INFO: 01-03 18:51:38: tf_logging.py:115 * 140451206297344 Restoring parameters from model/shadownet/shadownet_2019-01-03-18-33-13.ckpt-399
INFO: 01-03 18:51:40: demo_shadownet.py:114 * 140451206297344 Predict image test_01.jpg label 
DEBUG: 01-03 18:51:40: font_manager.py:1343 * 140451206297344 findfont: Matching :family=sans-serif:style=normal:variant=normal:weight=normal:stretch=normal:size=10.0 to DejaVu Sans ('/usr/local/lib/python2.7/dist-packages/matplotlib/mpl-data/fonts/ttf/DejaVuSans.ttf') with score of 0.050000


Anyone knows how to fix it?

**Comment by Antonio-hi** [comment]
请问一下这个问题你解决了吗？可以分享一下解决方案吗？
